### DISCUSSION TITLE
Single Workflow Concurrency

### DISCUSSION THREAD

**Discussion by AnassZaki** [community/community#26576] [Actions]
<p>I’ve just come across this new amazing feature <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#concurrency" class="inline-onebox" rel="noopener nofollow ugc">Workflow syntax for GitHub Actions - GitHub Docs</a> which is still in beta and I’m wondering if it’s possible to implement a method of controlling concurrency of one workflow. For example: the workflow is ran 5 times, we want to cancel all previous runs but keep the newest one; How would one achieve this?</p>

**Comment by Simran-B** [comment, marked as answer]
<p>At the top level of your workflow, add</p>
<pre><code class="lang-yaml">concurrency:
  group: workflow-xyz
  cancel-in-progress: true
</code></pre>
<p>You can use any string as group name, just make sure you don’t use it in any other workflow again. You may use <code>${{ github.workflow }}</code> which is the workflow name or file path.</p>
